styleguide-app/Examples/TextInput.elm

```diff
@@ -45,101 +45,113 @@ example =
-                        \onInput ->
-                            TextInput.newPassword
-                                { onInput = onInput
-                                , showPassword = state.showPassword
-                                , setShowPassword = SetShowPassword
-                                }
-                    , onFocus = "Focused!!!"
-                    , onBlur = "Blurred!!!"
-                    , onEnter = "Entered!!!"
-                    }
-                , toExample
-                    { name = "email"
-                    , toString = identity
-                    , inputType = TextInput.email
-                    , onFocus = "Focused!!!"
-                    , onBlur = "Blurred!!!"
-                    , onEnter = "Entered!!!"
-                    }
-                , toExample
-                    { name = "search"
-                    , toString = identity
-                    , inputType = TextInput.search
-                    , onFocus = "Focused!!!"
-                    , onBlur = "Blurred!!!"
-                    , onEnter = "Entered!!!"
-                    }
-                ]
+            , viewExamples <|
+                ( "readOnlyText"
+                , TextInput.view "Shareable Assignment Link"
+                    [ TextInput.readOnlyText
+                    , TextInput.value "noredink.com/s/blueprint-code"
+                    ]
+                )
+                    :: customizableExamples state
             ]
     }
 
 
+customizableExamples : State -> List ( String, Html Msg )
+customizableExamples state =
+    let
+        exampleConfig =
+            Control.currentValue state.control
+
+        toExample { name, toString, inputType, onFocus, onBlur, onEnter } index =
+            ( name
+            , TextInput.view exampleConfig.label
+                (exampleConfig.attributes
+                    ++ [ TextInput.id ("text-input__" ++ name ++ "-example")
+                       , inputType (toString >> SetInput index)
+                            |> TextInput.map toString identity
+                       , TextInput.value (Maybe.withDefault "" (Dict.get index state.inputValues))
+                       ]
+                    ++ List.filterMap identity
+                        [ if exampleConfig.onFocus then
+                            Just (TextInput.onFocus (SetInput index onFocus))
+
+                          else
+                            Nothing
+                        , if exampleConfig.onBlur then
+                            Just (TextInput.onBlur (SetInput index onBlur))
+
+                          else
+                            Nothing
+                        , if exampleConfig.onEnter then
+                            Just (TextInput.onEnter (SetInput index onEnter))
+
+                          else
+                            Nothing
+                        ]
+                )
+            )
+    in
+    List.indexedMap (\i toView -> toView i)
```

Comment:
NO CHANGE SUGGESTED. 

Just code golf: 
- `(\i toView -> toView i)` reduces to `(|>)` through identity
- while `(\toView i -> toView i)` reduces to `(<|)` through identity
- while `(\x -> someFunc x)` reduces to `someFunc` through eta reduction